### DISCUSSION TITLE
Outlined button groups

### DISCUSSION THREAD

**Discussion by SPodjasek** [twbs/bootstrap#38568] [Ideas]
Current implementation with negative margin sometimes renders badly - in my case on HiDPI displays. Some borders between buttons are visible as double width, maybe it's an issue with chrome but disabling left-side border on not-first elements solves this issue and probably won't break rendering in other cases.

Current implementation
https://github.com/twbs/bootstrap/blob/ea4da49f523de57872be0d701d8afbab002e475e/scss/_button-group.scss#L36-L43
looks like this:

![image](https://user-images.githubusercontent.com/3293161/236903913-9c33a9dc-d09f-4dfe-8814-03b38bba3b79.png)

After applying this hotfix:
```scss
.btn-group {
  // Prevent double borders when buttons are next to each other
  > :not(.btn-check:first-child) + .btn,
  > .btn-group:not(:first-child) {
    border-left: 0;
  }
}
```

![image](https://user-images.githubusercontent.com/3293161/236904234-2df40339-418c-486b-b80a-983e90a46b73.png)

Maybe it's worth implementing?

**Comment by julien-deramond** [comment]
Thanks for suggesting a hotfix @SPodjasek.
Would you be able to create a CodePen (or anything in the same spirit) containing the code used to render your example?
I'd like to reproduce the double-border thing by testing it with different combinations of OSs and browsers and try your patch.
Or maybe you've seen it as well in our documentation?

**Comment by SPodjasek** [comment]
I do see this in current documentation - https://getbootstrap.com/docs/5.3/components/button-group/

![image](https://user-images.githubusercontent.com/3293161/236931655-b9ffc6c4-2011-4b5b-b5fb-4601aa8d7b69.png)

![image](https://user-images.githubusercontent.com/3293161/236931854-8588c554-eabe-413e-8c1a-fb56729a0ee4.png)

![image](https://user-images.githubusercontent.com/3293161/236931963-0f6b522c-1366-406b-92f9-d9566f2998e5.png)

**Comment by SPodjasek** [comment]
It seems to be Chrome only, here's a screenshot from Firefox on the same machine

![image](https://user-images.githubusercontent.com/3293161/236932630-40dcfdec-e3d8-4a23-9b2f-5d1e4ee749f9.png)

My version of Chrome is 113.0.5672.63 and Firefox is 112.0.2

**Comment by SPodjasek** [comment]
I've found the root cause of this edge-case behavior. It only happens when you enable font scaling in system properties with non-integer values. In my case, I use Ubuntu 23.04 with Chrome on a 4K display, and my font scaling was 1,2x. It looks as it should when I set scaling to 1,0x or 2,0x. Also, it seems that Firefox doesn't respect system-wide font scaling.

**Reply by julien-deramond** [reply]
Haven't had the time to look at it precisely yet but your feedback is very interesting here. Based on this observation, do you think the patch suggested in the main comment of this discussion is still valid?

**Reply by SPodjasek** [reply]
IMHO disabling this border instead of moving margin 'back' by width of border gives the same visual result, and properly handles this rare edge-case that I've encountered. 

I don't know the reasoning behind currently implemented solution and because of that I wanted to start this as a discussion and not an issue.

**Reply by julien-deramond** [reply]
I've not checked all the renderings that could be impacted by such a modification, but here's a slight discrepancy that I can observe for example. I don't have the history of this specific change but I suppose that it can be linked.

Right now, when hovering the last item of https://twbs-bootstrap.netlify.app/docs/5.3/components/button-group/#basic-example:
<img width="364" alt="Screenshot 2023-05-13 at 11 18 07" src="https://github.com/twbs/bootstrap/assets/17381666/466f1630-ea91-4134-9cd3-b7462d103840">

With the modification, the left border is not displayed (obviously) for the "Middle" and "Right" buttons, contrary to the "Left" button which has a right border displayed. So it breaks the default rendering of Bootstrap and could have an impact as well for customization (to be studied
<img width="359" alt="Screenshot 2023-05-13 at 11 17 45" src="https://github.com/twbs/bootstrap/assets/17381666/20d99491-bee8-4309-beae-358f04fb1a97">
